### PR TITLE
[Fabric] Shim RCTComponentViewDescriptor

### LINKED ISSUE #1543: Shim RCTComponentViewDescriptor

### LINKED ISSUE #1559: Shim RCTSwitchComponentView

### PR DESCRIPTION
**Pull request by shwanton**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Fabric on macOS implementation:
Shim RCTComponentViewDescriptor to work w/ Fabric

closes #1543 

## Changelog

[macOS][Added] - Shim RCTComponentViewDescriptor for Fabric

## Test Plan

[x] Build RNTester-macOS w/ Fabric - doesn’t run yet, but no RCTComponentViewDescriptor errors
CleanShot 2023-01-23 at 21 20 49

[x] Build RNTester - iOS w/ Fabric
CleanShot 2023-01-23 at 21 08 15

[x] Build RNTester-macOS w/ Paper
CleanShot 2023-01-23 at 21 02 54

[x] Build RNTester - iOS w/ Paper
CleanShot 2023-01-23 at 21 01 28

### CODE REVIEW

File: React/Fabric/Mounting/RCTComponentViewDescriptor.h [hunk 1]
```diff
@@ -22,7 +22,7 @@ class RCTComponentViewDescriptor final {
    * Associated (and owned) native view instance.
    */
   __strong RCTUIView<RCTComponentViewProtocol> *view = nil; // [macOS]
-
+  NSInteger tag = 0; // default is 0 // [macOS]
```
Comment (Saadnajmi) [inline React/Fabric/Mounting/RCTComponentViewDescriptor.h:25]:
```suggestion
  NSInteger tag = 0; // [macOS] default is 0
```

Comment (Saadnajmi) [reply React/Fabric/Mounting/RCTComponentViewDescriptor.h:25]: Why does macOS need this but not iOS?

File: React/Fabric/Mounting/RCTComponentViewRegistry.mm [hunk 2]
```diff
@@ -83,7 +83,11 @@ - (void)preallocateViewComponents
       @"RCTComponentViewRegistry: Attempt to dequeue already registered component.");
 
   auto componentViewDescriptor = [self _dequeueComponentViewWithComponentHandle:componentHandle];
+#if !TARGET_OS_OSX // [macOS]
   componentViewDescriptor.view.tag = tag;
+#else // [macOS
+  componentViewDescriptor.tag = tag;
+#endif // macOS]
```
Comment (Saadnajmi) [inline React/Fabric/Mounting/RCTComponentViewRegistry.mm:90]: Have we not yet set a view yet on macOS, so instead we're directly setting the tag, presumably to a default value of 0 everywhere? Do we have an issue tracking setting it on the actual view? (either internally at Meta or publicly on our GH project)

**Review by Saadnajmi — APPROVED** [review]

**Review by Saadnajmi — COMMENTED** [review]
Couple of minor comments

File: React/Fabric/Mounting/RCTComponentViewDescriptor.h (hunk 1 above)
Comment (shwanton) [reply React/Fabric/Mounting/RCTComponentViewDescriptor.h:25]: This is temp shim till we merge our TouchHandler changes with proper fix

File: React/Fabric/Mounting/RCTComponentViewRegistry.mm (hunk 2 above)
Comment (shwanton) [reply React/Fabric/Mounting/RCTComponentViewRegistry.mm:90]: In the upcoming TouchHandler changes, this will be set to correct react tag.

**Pull request merged**